tools/querytee/proxy.go

```diff
@@ -151,13 +152,27 @@ func NewProxy(cfg ProxyConfig, logger log.Logger, routes []Route, registerer pro
 }
 
 func (p *Proxy) Start() error {
-	// Setup listener first, so we can fail early if the port is in use.
-	listener, err := net.Listen("tcp", fmt.Sprintf(":%d", p.cfg.ServerServicePort))
+	// Setup server first, so we can fail early if the ports are in use.
+	serv, err := server.New(server.Config{
```

Comment:
Other changes we have to do the config:
- MetricsNamespace: "cortex_querytee"
- RegisterInstrumentation: false
- ServerGracefulShutdownTimeout: 0
- GPRCServerMaxConcurrentStreams: 10000

Reply:
Noob question 😅  but where did you see those values?

And also based on `registerServerFlagsWithChangedDefaultValues`:
https://github.com/grafana/mimir/blob/640178b04ef28c4becc4985161b2599713b25273/pkg/mimir/mimir.go#L317-L318
Should we also set:
* GRPCServerMinTimeBetweenPings: 10s
* GRPCServerPingWithoutStreamAllowed: true

Reply:
Good point, yes please.